res/controllers/midi-components-0.0.js

```diff
@@ -318,21 +318,45 @@
         this.firstValueReceived = false;
     };
     Pot.prototype = new Component({
+        relative: false,
         inValueScale: function (value) { return value / this.max; },
-        input: function (channel, control, value, status, group) {
-            this.inSetParameter(this.inValueScale(value));
-            if (! this.firstValueReceived) {
-                this.firstValueReceived = true;
-                this.connect();
+        shift: function () {
+            if (this.relative) {
+                this.input = function (channel, control, value, status, group) {
+                    this.previousValueReceived = value;
+                }
+            }
+        },
+        unshift: function () {
+            this.input = function (channel, control, value, status, group) {
+                if (this.relative) {
+                    if (this.previousValueReceived !== undefined) {
+                        var delta = (value - this.previousValueReceived) / this.max;
+                        this.inSetParameter(this.inGetParameter() + delta);
```

Comment:
I have noticed some curious behavior testing with the `rate` ControlPotmeter. `engine.setParameter` allows setting out-of-bounds values if the ControlPotmeter allows it, but `engine.getParameter` clamps to 0...1. So, moving the slider to the top, then holding shift and moving it down, then moving it up increases it one step beyond the normal range, but not any further. That is odd, but not really bad. However, because `engine.getParameter` clamps to 0 ... 1, moving the slider jumps to the boundary of the normal range when the value of the ControlPotmeter is outside the range, which can happen with sync. That is bad.